### PR TITLE
Add boost 1.67.0

### PR DESCRIPTION
**Pull request by Rombur**
Add the latest version of boost. I had to set it as the preferred version, otherwise the release candidate was the preferred version.

### CODE REVIEW

**Comment by davydden** [comment]
@Rombur can you please check if `1.67.0` has the correct `array_binary_tree.hpp`? If not, we shall extend the patch from https://github.com/spack/spack/pull/7644/

**Comment by davydden** [comment]
p.s. the fix was merged upstream https://github.com/boostorg/graph/pull/104 but it might not make it to the release...

**Comment by Rombur** [comment]
It's there. There is no need for the patch,

**Comment by davydden** [comment]
> There is no need for the patch,

great!

File: var/spack/repos/builtin/packages/boost/package.py [hunk 1]
```diff
@@ -46,10 +46,13 @@ class Boost(Package):
             branch='develop',
             submodules=True)
 
+    version('1.67.0', '694ae3f4f899d1a80eb7a3b31b33be73c423c1ae',
+            url='https://dl.bintray.com/boostorg/release/1.67.0/source/boost_1_67_0.tar.bz2',
+            preferred=True)
     version('1.67.0.b1', '3423a4a3ec5297051fd27091864630e2dce4c159',
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/boost/package.py]: Now that `1.67.0` has been officially released, can we remove this beta version? Then we can remove `preferred=True` as well.

Comment (davydden) [reply var/spack/repos/builtin/packages/boost/package.py]: I agree, we can remove `beta` and its patch.

**Pull request merged**